### ISSUE TITLE
symbol's value as variable is void: slime-ritz

### ISSUE THREAD

**Issue by ronslow** [pallet/ritz#60]
I got this warning when installing slime-ritz using package-install.

And I get the message when auto-completing slime-

slime-connect hangs silently

I can't see this as an existing issue.

Any ideas?
Thanks

**Comment by mstump** [comment]
upgrading all of your elpa packages, and making sure that you're using the slime from melpa fixed this issue for me.

**Comment by arashbi** [comment]
I have the same problem with slime-ritz 0.4.2, is it the latest version?

**Comment by cola-zero** [comment]
Latest version is 0.5.0 [*](http://www.marmalade-repo.org/packages/slime-ritz).
If you have same problem with latest version, try recompile slime-ritz.el file.

**Comment by arashbi** [comment]
No luck. Compiling gives a bunch of warnings. and slime-connect gives the same error again.

-_\- mode: compilation; default-directory: "~/.emacs.d/elpa/slime-ritz-0.5.0/" -_-
Compilation started at Thu Nov 08 10:23:50

Compiling file c:/Users/abizhanzadeh@cta.loc/.emacs.d/elpa/slime-ritz-0.5.0/slime-ritz.el at Thu Nov 08 10:24:58 2012
slime-ritz.el:11:23:Warning: reference to free variable `slime-ritz'
slime-ritz.el:11:23:Warning:`:authors' called as a function
slime-ritz.el:13:4:Warning: `:license' called as a function
slime-ritz.el:14:4:Warning:`:on-load' called as a function
slime-ritz.el:17:16:Warning: reference to free variable `slime-repl-mode-map'
slime-ritz.el:19:16:Warning: reference to free variable`java-mode-map'

In slime-display-breakpoints:
slime-ritz.el:77:73:Warning: function `position' from cl package called at
    runtime
slime-ritz.el:187:3:Warning: function`remove_' from cl package called at
    runtime
slime-ritz.el:187:3:Warning: function `sort_' from cl package called at
    runtime

In slime-display-exception-filters:
slime-ritz.el:226:75:Warning: function `position' from cl package called at
    runtime
slime-ritz.el:319:3:Warning: function`remove_' from cl package called at
    runtime
slime-ritz.el:319:3:Warning: function `sort_' from cl package called at
    runtime

In slime-ritz-bind-repl-keys:
slime-ritz.el:460:15:Warning: reference to free variable `slime-repl-mode-map'

In end of data:
slime-ritz.el:468:1:Warning: the following functions are not known to be defined:
    define-slime-contrib, :authors, :license, :on-load,
    slime-buffer-name

**Comment by 11Kilobytes** [comment]
Yeah, I have the same thing here. Ritz isn't working for me as well and I get the same errors.

**Comment by hugoduncan** [comment]
Are these issues still occurring with 0.6.0?

**Comment by 11Kilobytes** [comment]
Yes, I am using version 0.6.0. Here's what I did:
From a fresh copy of Emacs 24. I added only MELPA to my package list using my .emacs file:

(require 'package)
(add-to-list 'package-archives
             '("melpa" . "http://melpa.milkbox.net/packages/") t)
(package-initialize)

I evaluated the buffer and did, M-x package-refresh-contents

Then I installed the slime version from the package provided you, by using M-x package-install-file.
After that, I launched a swank server using lein ritz in my project directory. When I try to connect to it with M-x slime-connect, I get the error, "slime-ritz-init: Symbol's value as variable is void: slime-ritz."

**Comment by 11Kilobytes** [comment]
I also get the same compilation errors as arashbi.

**Comment by arashbi** [comment]
cola,
Beside this problem, clojurescript has a very bad output in the nrepl in
general. I had to drop nrepl and ritz and go back to plain old
inferior-lisp mode and lein 1
Do you know anything in this regard?

**Comment by 11Kilobytes** [comment]
I think there might be an issue with versions. The slime version in the package that you can download from here is 20101113 but slime-ritz has a version of 20121115.

**Comment by cola-zero** [comment]
@arashbi,
I'm sorry but, I don't try clojurescript with nrepl yet, so I can say nothing about it.

@11Kilobytes,
Can you try `M-: (require 'slime-ritz)` in emacs?
I think this will fail and show some Backtrace.
I have similar problem with slime-fuzzy which is slime contrib module (and slime-ritz is also implemented as slime contrib module).
Here is content of Backtrace.
https://gist.github.com/4211629
If you have similar backtrace, try to remove slime-ritz.elc from load-path.

**Comment by 11Kilobytes** [comment]
I get this as the backtrace when I eval `(require 'slime-ritz)`

```
Debugger entered--Lisp error: (void-variable slime-ritz)
  byte-code("\304\305\306\307!\310\311!\312\313    \314\315#\313\n\314\315#\313      \316\317#\313\316\320#$%\207" [slime-ritz slime-mode-map slime-repl-mode-map java-mode-map define-slime-  contrib "Integration with ritz features" :authors "Hugo Duncan <hugo_duncan@yahoo.com>" :license "EPL" :on-load   define-key "" slime-break-on-exception "" slime-line-breakpoint slime-java-line-breakpoint] 13)
  require(slime-ritz)
  eval((require (quote slime-ritz)) nil)
  eval-expression((require (quote slime-ritz)) nil)
  call-interactively(eval-expression nil nil)
  recursive-edit()
```

**Comment by reidyp** [comment]
was this ever resolved ? I'm experiencing the same thing.

**Comment by 11Kilobytes** [comment]
I've made some progress, when you eval (require 'slime), slime-ritz.el and (require 'slime-ritz), with the version of slime-ritz in MELPA and slime from this Github, you are able to connect to a project, when you do Emacs hangs, so press an arrow key to interrupt whatever it's doing, what I find is that, I am able to go to a Clojure file, enable slime-mode and I can evaluate some expressions in the buffer, no REPL though.

**Comment by xiaohanyu** [comment]
Maybe this is the same problem like: https://bugs.launchpad.net/slime/+bug/1027361. 

And maybe add necessary `eval-when-compile` will solve this problem.
